Gems/Terrain/Code/Source/Components/TerrainWorldRendererComponent.cpp

```diff
@@ -58,19 +64,26 @@ namespace Terrain
                         ->Attribute(AZ::Edit::Attributes::Max, 10000.0f)
                     ;
 
+                editContext->Class<MeshConfiguration>("Mesh", "Settings related to rendering terrain meshes")
+                    ->DataElement(AZ::Edit::UIHandlers::Slider, &MeshConfiguration::m_renderDistance, "Mesh render distance", "The distance from the camera that terrain meshes will render.")
+                        ->Attribute(AZ::Edit::Attributes::Min, 1.0f)
+                        ->Attribute(AZ::Edit::Attributes::SoftMin, 100.0f)
+                        ->Attribute(AZ::Edit::Attributes::Max, 100000.0f)
+                        ->Attribute(AZ::Edit::Attributes::SoftMax, 10000.0f)
+                    ->DataElement(AZ::Edit::UIHandlers::Slider, &MeshConfiguration::m_firstLodDistance, "First LOD distance", "The distance from the camera that the first Lod renders to. Subsequent LODs will be at double the distance from the previous LOD.")
+                        ->Attribute(AZ::Edit::Attributes::Min, 1.0f)
+                        ->Attribute(AZ::Edit::Attributes::SoftMin, 10.0f)
+                        ->Attribute(AZ::Edit::Attributes::Max, 10000.0f)
+                        ->Attribute(AZ::Edit::Attributes::SoftMax, 1000.0f)
```

Comment:
Suggestion: Might want to consider rounding up to powers of two for the softmin/softmax/max values. I'm not sure that there is much savings to be gained from using non-powers-of-two here. I'm great with users being able to *select* any value, it just seems a bit weird to me for the limits to be sort of arbitrary. (Though admittedly maybe it's just because I've gotten too used to seeing power-of-two values for things)

Reply:
The number of mesh sectors per stack will be based on a combination of this number and the height query resolution. So there will be "sweet spots" you could say, where a range of number yields the same number of draws, but they won't necessarily be powers of two because query resolution can be anything. Also, we're clipping polygons in the shader based on these values, so even if it doesn't reduce draws, lowering the render distance or first LOD distance will still have a beneficial effect on performance.

Reply:
Yeah, that's totally fair. It just weirds me out to see non-power-of-two limits. :)